src/FileFormats.Minidump/Minidump.cs

```diff
@@ -0,0 +1,155 @@
+                    systemIndex = i;
+                }
+                else if (streamType == MINIDUMP_STREAM_TYPE.ModuleListStream)
+                {
+                    Debug.Assert(_moduleListStream == -1);
+                    _moduleListStream = i;
+                }
+            }
+
+            if (systemIndex == -1)
+                throw new BadInputFormatException("Minidump does not contain a MINIDUMP_SYSTEM_INFO stream");
+
+            _systemInfo = headerReader.Read<MINIDUMP_SYSTEM_INFO>(_position + _directory[systemIndex].Rva);
+
+            _dataSourceReader = new Reader(_dataSource, new LayoutManager().AddCrashDumpTypes(false, Is64Bit));
+            _loadedImages = new Lazy<List<MinidumpLoadedImage>>(CreateLoadedImageList);
+            _memoryRanges = new Lazy<List<MinidumpSegment>>(CreateSegmentList);
+            _virtualAddressReader = new Lazy<Reader>(CreateVirtualAddressReader);
+        }
+
+        /// <summary>
+        /// A raw data reader for the underlying minidump file itself.
+        /// </summary>
+        public Reader DataSourceReader { get { return _dataSourceReader; } }
+
+        /// <summary>
+        /// A raw data reader for the memory in virtual address space of this minidump.
+        /// </summary>
+        public Reader VirtualAddressReader { get { return _virtualAddressReader.Value; } }
+
+        /// <summary>
+        /// A collection of loaded images in the minidump.  This does NOT contain unloaded modules.
+        /// </summary>
+        public ReadOnlyCollection<MinidumpLoadedImage> LoadedImages { get { return _loadedImages.Value.AsReadOnly(); } }
+
+        /// <summary>
+        /// A collection of all the memory segments in minidump.
+        /// </summary>
+        public ReadOnlyCollection<MinidumpSegment> Segments { get { return _memoryRanges.Value.AsReadOnly(); } }
+
+        /// <summary>
+        /// Returns true if the original process represented by this minidump was running as an x64 process or not.
+        /// </summary>
+        public bool Is64Bit
+        {
+            get
+            {
+                var arch = _systemInfo.ProcessorArchitecture;
+                return arch == ProcessorArchitecture.PROCESSOR_ARCHITECTURE_ALPHA64 || arch == ProcessorArchitecture.PROCESSOR_ARCHITECTURE_AMD64 || arch == ProcessorArchitecture.PROCESSOR_ARCHITECTURE_IA64;
+            }
+        }
+
+        private Reader CreateVirtualAddressReader()
+        {
+            return _dataSourceReader.WithAddressSpace(new MinidumpVirtualAddressSpace(Segments, _dataSource));
+        }
+
+        private List<MinidumpLoadedImage> CreateLoadedImageList()
+        {
+            if (_moduleListStream == -1)
+                throw new BadInputFormatException("Minidump does not contain a ModuleStreamList in its directory.");
+            
+            MINIDUMP_MODULE[] modules = _dataSourceReader.ReadCountedArray<MINIDUMP_MODULE>(_directory[_moduleListStream].Rva);
```

Comment:
This wouldn't handle a non-zero position I think? Rather than remembering to add _position everywhere you can also create a new RelativeAddressSpace represents a portion of the original, and use that as the address space of the _dataSourceReader